grails-app/assets/javascripts/chartjsManager.js

```diff
@@ -288,18 +292,31 @@ function DashboardViewModel(config, searchFilters) {
     self.data = ko.observable()
     self.options = ko.observable(config.chartOptions.options)
 
+    if (self.options().scales != undefined) {
```

Comment:
do it such a way that it can be switched on via configuration. For example, add a config called `chartOptions.minifyXLabel` which accepts a number. If number is greater than 0, then that will turn on this feature and also will provide the value for `maxLabelLength`.